src/test/java/io/confluent/connect/elasticsearch/ElasticsearchWriterTest.java

```diff
@@ -134,8 +136,33 @@ public void testCompatible() throws Exception {
     records.add(sinkRecord);
     expected.add(sinkRecord);
 
-    writeDataAndWait(writer, records, SLEEP_INTERVAL_MS);
-    verifySearch(expected, search(client), true);
+    writeDataAndRefresh(writer, records);
+    verifySearchResults(expected, true);
+  }
+
+  @Test
+  public void testSafeRedelivery() throws Exception {
+    final boolean ignoreKey = false;
+
+    final Struct value0 = new Struct(schema);
+    value0.put("user", "foo");
+    value0.put("message", "hi");
+    final SinkRecord sinkRecord0 = new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, key, schema, value0, 0);
+
+    final Struct value1 = new Struct(schema);
+    value1.put("user", "foo");
+    value1.put("message", "bye");
+    final SinkRecord sinkRecord1 = new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, key, schema, value1, 1);
+
+    final ElasticsearchWriter writer = initWriter(client, ignoreKey, false, Collections.<String, TopicConfig>emptyMap());
+    writer.write(Arrays.asList(sinkRecord0, sinkRecord1));
+    writer.flush();
+
+    // write the record with earlier offset again
+    writeDataAndRefresh(writer, Collections.singleton(sinkRecord0));
+
+    // last write should have been ignored due to version conflict
+    verifySearchResults(Collections.singleton(sinkRecord1), ignoreKey);
```

Comment:
Is this actually a sufficient test given how `verifySearchResults` works? It only validates that the IDs match, but the IDs for these documents will be the same. This test passes even if you pass `sinkRecord0` here instead of `sinkRecord1`.

The version is probably returned, so to properly validate I think you'd really want a variant of `verifySearchResults` where you can pass expected versions in. (And arguably might want to validate both before and after the second write.)